interp/src/interpret_control.rs

```diff
@@ -0,0 +1,171 @@
+//! Inteprets a control in a component.
+
+use super::{
+    environment::Environment, interpret_group::GroupInterpreter, interpreter,
+};
+use calyx::{
+    errors::{Error, FutilResult},
+    ir,
+};
+use std::rc::Rc;
+
+/// Interpret a control.
+/// TODO: Attributes, implementation for some component variants
+pub struct ControlInterpreter {
+    /// The environment
+    pub environment: Environment,
```

Comment:
Feels strange to me that the interpreter owns the environment, rather than having the environment passed in. Interpreting any control statement should be largely stateless from the perspective of the control interpreter, pass in env get out env.